src/document/printer.js

```diff
@@ -142,27 +142,41 @@ function generateInd(ind, newPart, options) {
   }
 }
 
-// TODO: Skip cursor placeholder
 // Trim `Tab(U+0009)` and `Space(U+0020)` at the end of line
 function trim(out) {
   let trimCount = 0;
+  let cursorCount = 0;
+  let outIndex = out.length;
 
-  while (out.length > 0) {
-    const last = getLast(out);
-    if (typeof last !== "string") {
-      break;
+  outer: while (outIndex--) {
+    const last = out[outIndex];
+
+    if (last === CURSOR_PLACEHOLDER) {
+      cursorCount++;
```

Comment:
I'd use an array to save cursors, and push them directly after text trimmed. But use number is fine too.